From 7bb7dc3679afe3aa4957a69aa93d443525b85a3e Mon Sep 17 00:00:00 2001
From: "Patil, Rachna" <rachna@ti.com>
Date: Tue, 8 May 2012 10:57:12 +0530
Subject: [PATCH 0950/1092] input: TSC: Remove support for TSC on alpha EVM

4 wire analog inputs on alpha EVM's were interchanged,
Alpha AM335x EVM's are not supported anymore.
Code related to this is dropped.

Signed-off-by: Patil, Rachna <rachna@ti.com>
---
 drivers/input/touchscreen/ti_tscadc.c | 60 +++++++----------------------------
 include/linux/input/ti_tscadc.h       |  9 ------
 2 files changed, 12 insertions(+), 57 deletions(-)

diff --git a/drivers/input/touchscreen/ti_tscadc.c b/drivers/input/touchscreen/ti_tscadc.c
index 835a0ba..d9c13be 100644
--- a/drivers/input/touchscreen/ti_tscadc.c
+++ b/drivers/input/touchscreen/ti_tscadc.c
@@ -107,7 +107,6 @@ unsigned int bckup_x = 0, bckup_y = 0;
 struct tscadc {
 	struct input_dev	*input;
 	int			wires;
-	int			analog_input;
 	int			x_plate_resistance;
 	int			irq;
 	void __iomem		*tsc_base;
@@ -141,29 +140,16 @@ static void tsc_step_config(struct tscadc *ts_dev)
 
 	switch (ts_dev->wires) {
 	case 4:
-		if (ts_dev->analog_input == 0)
-			stepconfigx |= TSCADC_STEPCONFIG_INP_4 |
-				TSCADC_STEPCONFIG_YPN;
-		else
-			stepconfigx |= TSCADC_STEPCONFIG_INP |
+		stepconfigx |= TSCADC_STEPCONFIG_INP |
 				TSCADC_STEPCONFIG_XNN;
 		break;
 	case 5:
 		stepconfigx |= TSCADC_STEPCONFIG_YNN |
-				TSCADC_STEPCONFIG_INP_5;
-		if (ts_dev->analog_input == 0)
-			stepconfigx |= TSCADC_STEPCONFIG_XNP |
-				TSCADC_STEPCONFIG_YPN;
-		else
-			stepconfigx |= TSCADC_STEPCONFIG_XNN |
+				TSCADC_STEPCONFIG_INP_5 | TSCADC_STEPCONFIG_XNN |
 				TSCADC_STEPCONFIG_YPP;
 		break;
 	case 8:
-		if (ts_dev->analog_input == 0)
-			stepconfigx |= TSCADC_STEPCONFIG_INP_4 |
-				TSCADC_STEPCONFIG_YPN;
-		else
-			stepconfigx |= TSCADC_STEPCONFIG_INP |
+		stepconfigx |= TSCADC_STEPCONFIG_INP |
 				TSCADC_STEPCONFIG_XNN;
 		break;
 	}
@@ -178,25 +164,14 @@ static void tsc_step_config(struct tscadc *ts_dev)
 			TSCADC_STEPCONFIG_INM | TSCADC_STEPCONFIG_FIFO1;
 	switch (ts_dev->wires) {
 	case 4:
-		if (ts_dev->analog_input == 0)
-			stepconfigy |= TSCADC_STEPCONFIG_XNP;
-		else
-			stepconfigy |= TSCADC_STEPCONFIG_YPP;
+		stepconfigy |= TSCADC_STEPCONFIG_YPP;
 		break;
 	case 5:
-		stepconfigy |= TSCADC_STEPCONFIG_XPP | TSCADC_STEPCONFIG_INP_5;
-		if (ts_dev->analog_input == 0)
-			stepconfigy |= TSCADC_STEPCONFIG_XNN |
-				TSCADC_STEPCONFIG_YPP;
-		else
-			stepconfigy |= TSCADC_STEPCONFIG_XNP |
-				TSCADC_STEPCONFIG_YPN;
+		stepconfigy |= TSCADC_STEPCONFIG_XPP | TSCADC_STEPCONFIG_INP_5 |
+			TSCADC_STEPCONFIG_XNP | TSCADC_STEPCONFIG_YPN;
 		break;
 	case 8:
-		if (ts_dev->analog_input == 0)
-			stepconfigy |= TSCADC_STEPCONFIG_XNP;
-		else
-			stepconfigy |= TSCADC_STEPCONFIG_YPP;
+		stepconfigy |= TSCADC_STEPCONFIG_YPP;
 		break;
 	}
 
@@ -205,15 +180,9 @@ static void tsc_step_config(struct tscadc *ts_dev)
 		tscadc_writel(ts_dev, TSCADC_REG_STEPDELAY(i), delay);
 	}
 
-	chargeconfig = TSCADC_STEPCONFIG_XPP |
-			TSCADC_STEPCONFIG_YNN |
-			TSCADC_STEPCONFIG_RFP |
-			TSCADC_STEPCHARGE_RFM;
-	if (ts_dev->analog_input == 0)
-		chargeconfig |= TSCADC_STEPCHARGE_INM_SWAP |
-			TSCADC_STEPCHARGE_INP_SWAP;
-	else
-		chargeconfig |= TSCADC_STEPCHARGE_INM | TSCADC_STEPCHARGE_INP;
+	chargeconfig = TSCADC_STEPCONFIG_XPP | TSCADC_STEPCONFIG_YNN |
+			TSCADC_STEPCONFIG_RFP | TSCADC_STEPCHARGE_RFM |
+			TSCADC_STEPCHARGE_INM | TSCADC_STEPCHARGE_INP;
 	tscadc_writel(ts_dev, TSCADC_REG_CHARGECONFIG, chargeconfig);
 	tscadc_writel(ts_dev, TSCADC_REG_CHARGEDELAY, TSCADC_STEPCHARGE_DELAY);
 
@@ -237,12 +206,8 @@ static void tsc_idle_config(struct tscadc *ts_config)
 	/* Idle mode touch screen config */
 	unsigned int	 idleconfig;
 
-	idleconfig = TSCADC_STEPCONFIG_YNN |
-			TSCADC_STEPCONFIG_INM | TSCADC_STEPCONFIG_IDLE_INP;
-	if (ts_config->analog_input == 0)
-		idleconfig |= TSCADC_STEPCONFIG_XNN;
-	else
-		idleconfig |= TSCADC_STEPCONFIG_YPN;
+	idleconfig = TSCADC_STEPCONFIG_YNN | TSCADC_STEPCONFIG_INM |
+			TSCADC_STEPCONFIG_IDLE_INP | TSCADC_STEPCONFIG_YPN;
 
 	tscadc_writel(ts_config, TSCADC_REG_IDLECONFIG, idleconfig);
 }
@@ -456,7 +421,6 @@ static	int __devinit tscadc_probe(struct platform_device *pdev)
 	tscadc_writel(ts_dev, TSCADC_REG_CLKDIV, clk_value);
 
 	ts_dev->wires = pdata->wires;
-	ts_dev->analog_input = pdata->analog_input;
 	ts_dev->x_plate_resistance = pdata->x_plate_resistance;
 
 	/* Set the control register bits */
diff --git a/include/linux/input/ti_tscadc.h b/include/linux/input/ti_tscadc.h
index 2c547bb..57ce71c 100644
--- a/include/linux/input/ti_tscadc.h
+++ b/include/linux/input/ti_tscadc.h
@@ -3,19 +3,10 @@
  * @wires:		Wires refer to application modes
  *			i.e. 4/5/8 wire touchscreen support
  *			on the platform
- * @analog_input:	Analog inputs refer to the order in which the
- *			connections are made to the AFE. If the connections
- *			are as : XPUL = AN0,XNUR = AN1,YPLL = AN2,
- *			YNLR = AN3, then this variable is set to 1.
- *			Else if the connections are as :
- *			XPUL = AN0,YPLL = AN1,XNUR = AN2,
- *			YNLR = AN3, then set this variable to
- *			0.
  * @x_plate_resistance:	X plate resistance.
  */
 
 struct tsc_data {
 	int wires;
-	int analog_input;
 	int x_plate_resistance;
 };
-- 
1.7.11.2

